test/consumer.integration.spec.ts

```diff
@@ -188,7 +192,9 @@ describe('FFI integration test for the HTTP Consumer API', () => {
       interaction.withQuery('someParam', 0, 'someValue');
       interaction.withRequestBinaryBody(
         bytes,
-        isWin ? 'application/octet-stream' : 'application/gzip'
```

Comment:
I realise this was already this way when you got here, but there should not be if statements in tests - *especially* on platform.

Reply:
Totally accept that and agreed, it points to a wider issue we have with content matching both cross platform and cross arch.

It's important to have these tests in, rather than skipping them, but raising these as discussion points so we can either fix the issue at source in the rust core, or provide platform/arch specific guidance where it is required.

This is a smell here, that is going to affect end users, unless they are aware of it

Reply:
failing making this change, there are just failing tests for something unrelated to the contributors change, which leads to a sub-optimal experience all round :(

Reply:
all fixed now in the rust core, and resolved in pact-js-core/pact-js (and other consuming libraries)

https://github.com/pact-foundation/pact-reference/pull/429